tokio/src/runtime/basic_scheduler.rs

```diff
@@ -293,6 +295,10 @@ where
                 self.local.park.park().ok().expect("park failed");
             }
         }
+
+        // Remove any budgeting constraints in case later code
+        // on the same thread decides to _also_ poll futures
+        crate::coop::opt_out();
```

Comment:
Should opting in / out be tied to the runtime instead of individual schedulers?

Reply:
It looks like you already have "opt_in" in `enter.rs`, maybe opt-out should be there?

Reply:
I did not place `opt_out` there, because some runtimes need to opt out earlier than that. For example, the threaded runtime opts out before it starts shutdown.